src/read.rs

```diff
@@ -110,49 +110,51 @@ where
         &mut self,
         mut n: usize,
         scratch: &mut Vec<u8>,
-        mut scratch_offset: usize,
+        scratch_offset: usize,
     ) -> Result<Reference<'de>> {
+        assert!(scratch.len() == scratch_offset);
```

Comment:
Yes, at least while the Deserializer uses it as it does now.

I haven't changed the internal API part yet because there are two ways of going forward here (drop scratch_offset, or drop the buf.clear() in the reader and make scratch_offset a boolean "start anew"), see https://github.com/pyfisch/cbor/pull/80/commits/d1903ffe2a0a30b66d693348fa4f4978f171d938 -- but I think those are better explored in a separate change.

Reply:
Ok cool - it seems fine to follow up in a separate change.